### PR TITLE
[Doc] Update dataclass.md

### PR DESCRIPTION
**Pull request by vai9er**
Issue: #

### CODE REVIEW

File: docs/lang/articles/advanced/dataclass.md [hunk 1]
```diff
@@ -22,19 +22,16 @@ class Sphere:
...
-The `@ti.dataclass` decorator converts the annotated members in the Python class to members in the resulting struct type. In both of the above examples, you end up with the same struct field.
+The `@ti.dataclass` decorator converts the annotated members in the `Python class` to members in the resulting `Struct Type`. In both of the above examples, you end up with the same struct field.
 
-```python
-sphere_field = Sphere.field(shape=(n,))
-```
 
-## Associate functions with the struct type
-Python classes can have functions attached to them, and so can Taichi struct types. Building from the above example, one can embed functions in the struct as follows:
+## Associate Functions with the Struct Type
```
Comment (neozhaoliang) [inline docs/lang/articles/advanced/dataclass.md:33]:
```suggestion
## Associate Functions with the Struct Type

```

Let's leave a blank line here for visual clarity.

File: docs/lang/articles/advanced/dataclass.md [hunk 2]
```diff
@@ -22,19 +22,16 @@ class Sphere:
...
     radius: ti.f32
 ```
-This creates the *exact* same type as using `ti.types.struct()`:
+This is the same equivalent as using `ti.types.struct()`:
 
 ```python
 Sphere = ti.types.struct(center=vec3, radius=ti.f32)
 ```
-The `@ti.dataclass` decorator converts the annotated members in the Python class to members in the resulting struct type. In both of the above examples, you end up with the same struct field.
+The `@ti.dataclass` decorator converts the annotated members in the `Python class` to members in the resulting `Struct Type`. In both of the above examples, you end up with the same struct field.
```
Comment (Olinaaaloompa) [inline docs/lang/articles/advanced/dataclass.md:30]: @neozhaoliang @vai9er Can you check the use of the inline code format here? I thought inline code should be used for excerpts from code snippets.

File: docs/lang/articles/advanced/dataclass.md [hunk 3]
```diff
@@ -6,13 +6,13 @@ sidebar_position: 4
...
-- A Python representation of the struct type which is more object oriented.
-- Functions associated with a struct type (C++-style structs).
+- A Python representation of the struct type which is more Object-Oriented.
+- Functions associated with a struct type.
 
 
 To achieve the ends, Taichi enabled the `@ti.dataclass` decorator on a Python class. This is inspired by Python's [dataclass](https://docs.python.org/3/library/dataclasses.html) feature, which uses class fields with annotations to create data types.
 
-## Create a struct from a Python class
+## Create a Struct from a Python Class
```
Comment (Olinaaaloompa) [inline docs/lang/articles/advanced/dataclass.md:15]: Let's use sentence case for subheadings :)

**Review by Olinaaaloompa — CHANGES_REQUESTED** [review]

File: docs/lang/articles/advanced/dataclass.md (hunk 3 above)
Comment (vai9er) [reply docs/lang/articles/advanced/dataclass.md:15]: I think headings should have the relevant words capitalized. We will discuss in upcoming meeting

Comment (Olinaaaloompa) [inline docs/lang/articles/advanced/dataclass.md:15]:
```suggestion
## Create a struct from a Python class
```

File: docs/lang/articles/advanced/dataclass.md [hunk 4]
```diff
@@ -22,19 +22,17 @@ class Sphere:
...
-The `@ti.dataclass` decorator converts the annotated members in the Python class to members in the resulting struct type. In both of the above examples, you end up with the same struct field.
+The `@ti.dataclass` decorator converts the annotated members in the `Python class` to members in the resulting `Struct Type`. In both of the above examples, you end up with the same struct field.
 
-```python
-sphere_field = Sphere.field(shape=(n,))
-```
 
-## Associate functions with the struct type
-Python classes can have functions attached to them, and so can Taichi struct types. Building from the above example, one can embed functions in the struct as follows:
+## Associate Functions with the Struct Type
```
Comment (Olinaaaloompa) [inline docs/lang/articles/advanced/dataclass.md:33]:
```suggestion
## Associate Functions with the struct type
```

**Review by Olinaaaloompa — CHANGES_REQUESTED** [review]

File: docs/lang/articles/advanced/dataclass.md [hunk 5]
```diff
@@ -22,19 +22,17 @@ class Sphere:
...
     radius: ti.f32
 ```
-This creates the *exact* same type as using `ti.types.struct()`:
+This is the same equivalent as using `ti.types.struct()`:
 
 ```python
 Sphere = ti.types.struct(center=vec3, radius=ti.f32)
 ```
-The `@ti.dataclass` decorator converts the annotated members in the Python class to members in the resulting struct type. In both of the above examples, you end up with the same struct field.
+The `@ti.dataclass` decorator converts the annotated members in the `Python class` to members in the resulting `Struct Type`. In both of the above examples, you end up with the same struct field.
```
Comment (Olinaaaloompa) [inline docs/lang/articles/advanced/dataclass.md:30]:
```suggestion
The `@ti.dataclass` decorator converts the annotated members in the *Python class* to members in the resulting *struct type*. In both of the above examples, you end up with the same struct field.
```

**Pull request merged**